runtime/src/accounts_db.rs

```diff
@@ -2715,12 +2697,53 @@ impl AccountsDB {
         }
     }
 
-    fn store_with_hashes(&self, slot: Slot, accounts: &[(&Pubkey, &Account)], hashes: &[Hash]) {
+    fn store_accounts_default<'a>(
+        &'a self,
+        slot: Slot,
+        accounts: &[(&Pubkey, &Account)],
+        hashes: &[Hash],
+    ) {
+        self.store_accounts_custom(
+            slot,
+            accounts,
+            hashes,
+            None::<StorageFinder>,
+            None::<Box<dyn Iterator<Item = u64>>>,
+        );
+    }
+
+    fn store_accounts_custom<'a>(
+        &'a self,
+        slot: Slot,
+        accounts: &[(&Pubkey, &Account)],
+        hashes: &[Hash],
+        storage_finder: Option<StorageFinder<'a>>,
+        write_version_producer: Option<Box<dyn Iterator<Item = u64>>>,
+    ) -> StoreAccountsTiming {
+        let storage_finder: StorageFinder<'a> = storage_finder
+            .unwrap_or_else(|| Box::new(move |slot, size| self.find_storage_candidate(slot, size)));
```

Comment:
I think we can avoid `unwrap` here by moving this into its sole call-site: `store_accounts_default`. Also, this makes `store_accounts_default` hold the definition of default behavior, which is easier to read.

And, `store_account_custom` now can take `StorageFinder<'a>` not wrapped in `Option<_>`. :)

Reply:
yeah this would be true based only on this PR 😄 , but I'm actually using these options in a few places in my other work here:

https://github.com/solana-labs/solana/pull/13140/files#diff-1090394420d51617f3233275c2b65ed706b35b53b115fe65f82c682af8134a6fR1178-R1179

https://github.com/solana-labs/solana/pull/13140/files#diff-1090394420d51617f3233275c2b65ed706b35b53b115fe65f82c682af8134a6fR2184-R2185

https://github.com/solana-labs/solana/pull/13140/files#diff-1090394420d51617f3233275c2b65ed706b35b53b115fe65f82c682af8134a6fR2994-R2995